src/ol/render/canvas/textreplay.js

```diff
@@ -304,7 +304,13 @@ ol.render.canvas.TextReplay.prototype.getImage_ = function(text, fill, stroke) {
 
   var lines = text.split('\n');
   var numLines = lines.length;
-  if (!ol.render.canvas.TextReplay.labelCache_.containsKey(key)) {
+  var hasLabel = ol.render.canvas.TextReplay.labelCache_.containsKey(key);
+  var checkLabel = hasLabel && !this.labelsChecked_;
+  if (!hasLabel || checkLabel) {
+    if (checkLabel) {
+      label = ol.render.canvas.TextReplay.labelCache_.get(key);
+      this.labelsChecked_ = true;
+    }
```

Comment:
This looks like we only check once.  I'm confused how this works if the font loads after this check.

Reply:
We check when a new replay is created, which is when a new tile is loaded or a new resolution for an existing tile is used.

Do you think we should run the check once per render frame instead?

Reply:
Curious if it would be possible to use document.fonts.check() to avoid the check and ready() to render after load.  https://developer.mozilla.org/en-US/docs/Web/API/CSS_Font_Loading_API

Reply:
Good hint. I'll check that.

Reply:
Unfortunately that `document.fonts` API is unavailable in Internet Explorer and Edge, and gives false positives in Safari. But I rewrote the whole pull request to use a simple custom check for available fonts, and to re-render as soon as a font is loaded. This should now even work better than in previous versions, where at least an interaction or animation was required to update the view when fonts became available.